chew.go

```diff
@@ -201,3 +424,102 @@ func processURL(url string, ctxs ...context.Context) ([]common.Chunk, error) {
 
 	return processor(resp.Body, url)
 }
+
+func getRobotsTxtInfo(urlStr string) (bool, time.Duration, error) {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return false, 0, err
+	}
+
+	robotsURL := fmt.Sprintf("%s://%s/robots.txt", parsedURL.Scheme, parsedURL.Host)
+
+	robotsMu.RLock()
+	robotsData, exists := robotsCache[robotsURL]
+	robotsMu.RUnlock()
+
+	/*
+		This bit of code makes a lot of very unhealthy assumptions, in an ideal world this
+		would be more straightforward but unfortunately it's not.
+
+		Use this with caution and respect for site owners.
+	*/
+	if !exists {
+		resp, err := http.Get(robotsURL)
+		if err != nil {
+			return true, crawlDelay, nil
```

Comment:
returning a crawl delay via this function while  crawlDelay is a global variable is a bad idea

I mean this function don't handle/alter the crawlDelay, you should simply use it where you call this function